### DISCUSSION TITLE
Data push from Jaeger to Skywalking Zipkin receiver

### DISCUSSION THREAD

**Discussion by roshnithadevus27** [apache/skywalking#11856] [Q&A] [answered]
Hi,

OAP Version -> 9.7
Jaeger Version -> 1.53

Configuration of Zipkin receiver in Application.yaml

<img width="649" alt="Screenshot 2024-02-01 115901" src="https://github.com/apache/skywalking/assets/156880868/e29d272d-2297-45f5-8fe6-2fad22f97080">

Our Jaeger server is running on a different VM, and we have tried to give the IP before the port in the Jaeger command which is giving this error. so, what procedure do we need to follow to push the data to Skywalking using the Zipkin receiver? 

The command is used to run the Jaeger collector.

![MicrosoftTeams-image (18)](https://github.com/apache/skywalking/assets/156880868/bf4df32e-85c5-4317-a5e3-d62510c7b1a3)

We tried to manually push data to the Zipkin receiver using API via Postman. We have gotten 202 successes but not reflected in the Skywalking lens UI.

![MicrosoftTeams-image (20)](https://github.com/apache/skywalking/assets/156880868/b3024f0e-4ec7-46ac-a631-ac2fe21d8eca)

![MicrosoftTeams-image (19)](https://github.com/apache/skywalking/assets/156880868/e345a01c-d34e-4a9e-9834-8c62d330e869)

**Comment by wu-sheng** [comment]
About postman part, if you could provide complete payload fot the HTTP, @wankai123 could help.

About Jaeger, I am not sure how it reports traces in Zipkin mode. Is that supported well by Jaeger SDKs?

**Reply by Maurya243** [reply]
Thank you for your prompt response.
I have provided the complete payload below. If I need to pass data from Jaeger currently, we are trying to push the data from jaeger using the Zipkin compatible rest API to Zipkin receiver in skywalking. ( https://www.jaegertracing.io/docs/1.53/getting-started/ ) . You can see in the migrating from Zipkin section.
Is there any other way we can push jaeger data to skywalking ?

**Reply by wu-sheng** [reply]
Another suggestion is using OTEL collector to convert the format.

We will check and see why Jaeger implementation about zipkin format doesn't work out.

**Reply by Maurya243** [reply]
Thanks for the response. 
Is there any sample OTEL yml that I can try to test the same.I think I need to use the OTEL contrib version to push to skywalking is it?

**Reply by wu-sheng** [reply]
I don't have details, but they have Jaeger and Zipkin support for sure.

**Comment by wankai123** [comment]
What is the input timestamp of this trace, and does it match your query time?

**Reply by wankai123** [reply]
It seems much more than 12h according to your query endTs.

**Reply by Maurya243** [reply]
Below is the complete payload of the postman. We have updated the time on both Zipkin and postman.
[
  {
    "traceId": "1a2b3c4d5e6f7890",
    "id": "a1b2c3d4e5f67890",
    "parentId": "0123456789abcdef",
    "name": "example-span",
    "timestamp": 1706849360,
    "duration": 500000,
    "localEndpoint": {
      "serviceName": "example-service",
      "ipv4": "127.0.0.1",
      "port": 8080
    },
    "remoteEndpoint": {
      "serviceName": "remote-service",
      "ipv4": "192.168.1.1",
      "port": 9090
    },
    "annotations": [
      {
        "timestamp": 1706849360,
        "value": "span-start"
      },
      {
        "timestamp": 1706849360,
        "value": "span-end"
      }
    ],
    "tags": {
      "tag-key-1": "tag-value-1",
      "tag-key-2": "tag-value-2"
    }
  }
]

**Reply by wankai123** [reply, marked as answer]
your trace data timestamp is not `us` format, try using `1706849360000000` instead

**Reply by wankai123** [reply]
It works for me.
<img width="1700" alt="image" src="https://github.com/apache/skywalking/assets/16773043/7367c51c-7254-4d38-a098-68d647bf2e38">

**Reply by Maurya243** [reply]
Thank you for your response. 
By changing the format as suggested we are now getting the trace in Zipkin UI .
Is it possible to showcase the same in the Sky waking UI. I have seen something in the UI template mesh part where it is showing the same data. But in my case, it was not coming as a service in Sky Waling UI

**Reply by wankai123** [reply]
> Is it possible to showcase the same in the Sky waking UI. I have seen something in the UI template mesh part where it is showing the same data. But in my case, it was not coming as a service in Sky Waling UI

I'm not sure what's the meaning, could you provide more details?

**Reply by Maurya243** [reply]
The data coming in Zipkin UI, same we need to showcase in the sky walking UI. Is there any steps or procedure to show the data in Skywalking UI

**Reply by wankai123** [reply]
If you mean to show the Zipkin data on the SkyWalking trace UI, they can't for now.  Zipkin trace format and mechanism are different from SkyWalking, Zipkin trace can't trans to Skywalking trace and SkyWalking doesn't do additional analysis of the Zipkin data.

**Reply by Maurya243** [reply]
Adding one more query is that by the below link they were suggesting that we could create a widget in Sky walking to show entire Zipkin UI. Is it available in the latest version 9.7. If yes, could you please suggest the steps on how to configure the same.
https://github.com/apache/skywalking/issues/10000

**Reply by wankai123** [reply]
SkyWalking UI already integrates Zipkin lens-ui, under menu service mesh -> services.
<img width="1187" alt="image" src="https://github.com/apache/skywalking/assets/16773043/96fb7db8-d622-4b30-9271-de889529ac62">

If you want to add Zipkin lens-ui widget in other places：
1.  refer to the doc to add a new widget.
2. the widget type select add iframe.
<img width="211" alt="image" src="https://github.com/apache/skywalking/assets/16773043/21d078e2-d641-4152-909f-d0108b9ab81f">

3. edit this widget and set the link to `/zipkin`
<img width="573" alt="image" src="https://github.com/apache/skywalking/assets/16773043/31ff2392-93e3-464c-8374-ae2596f346bd">

**Reply by Maurya243** [reply]
But Service mesh is not coming in skywalking UI. We are able to add iframe and show the Zipkin UI but ideally if data is there in Zipkin UI service mesh should be visible in Skywalking UI.

**Reply by wu-sheng** [reply]
No, it wouldn't until you send ALS or metrics service to OAP.

**Reply by wu-sheng** [reply]
Zipkin bundled UI is an add-on, it can't interact like other native data.

**Reply by wu-sheng** [reply]
You could use marketplace and click into mesh to check the UI.

For use cases
- If you need mesh observability, als is recommended, so UI is good
- If you don't need more, you could deploy Zipkin Lens UI images, and connect with OAP
